app/src/routes/login/components/sso-links.vue

```diff
@@ -55,7 +57,7 @@ export default defineComponent({
 						link: `${getRootPath()}auth/login/${provider.name}?redirect=${window.location.href.replace(
 							location.search,
```

Comment:
Did some more testing, and the true problem you're trying to solve is this particular line! This problem can be fixed without any additional API logic in the OAuth2 driver 👍🏻